airflow/utils/log/secrets_masker.py

```diff
@@ -143,6 +144,10 @@ def _is_v1_env_var(v: Any) -> TypeGuard[V1EnvVar]:
     return isinstance(v, _get_v1_env_var_type())
 
 
+def _is_state_enum(v: Any) -> bool:
+    return isinstance(v, (TaskInstanceState, DagRunState, JobState, State))
```

Comment:
Would like to understand if the secrets masker fails for all kinds of Enum values and if yes, would it be safe to have here,
isinstance(item, Enum)

Reply:
Yeah, that's correct. I will change it to Enum